src/msg/async/AsyncConnection.cc

```diff
@@ -811,3 +817,59 @@ void AsyncConnection::tick(uint64_t id)
     }
   }
 }
+
+void AsyncConnection::dump(Formatter *f) {
+  std::lock_guard<std::mutex> l(lock);
+
+  f->open_object_section("async_connection");
+  f->dump_string("state", get_state_name(state));
+  f->dump_unsigned("messenger_nonce", async_msgr->get_nonce());
+
+  f->open_object_section("status");
+  f->dump_bool("connected", is_connected());
+  f->dump_bool("loopback", is_loopback);
+  f->close_section();
+
+  if (cs) {
+    f->dump_int("socket_fd", cs.fd());
+    if (!dump_tcp_info(cs.fd(), f)) {
+      f->dump_null("tcp_info");
+    }
+  } else {
+    f->dump_null("socket_fd");
+    f->dump_null("tcp_info");
+  }
+  f->dump_int("conn_id", conn_id);
+
+  f->open_object_section("peer");
+  f->dump_string("entity_name", get_peer_entity_name().to_str());
```

Comment:
```suggestion
  f->dump_object("entity_name", get_peer_entity_name());
```

Let's avoid stringifying any objects unnecessarily. Please check other fields as well.